modin/pandas/base.py

```diff
@@ -122,6 +122,23 @@ class BasePandasDataset(ClassLogger):
     # but lives in "pandas" namespace.
     _pandas_class = pandas.core.generic.NDFrame
 
+    def _is_dataframe(self) -> bool:
```

Comment:
I also suggest decorating this with `pandas.util.cache_readonly` (this would turn this method in a once-called property)

Reply:
Done.